File: backend/models/activity.py
import uuid
from sqlalchemy import Column, String, DateTime, ForeignKey, Float, Integer, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from backend.database import Base

class Activity(Base):
    __tablename__ = "activities"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    trip_id = Column(String(36), ForeignKey("trips.id", ondelete="CASCADE"), nullable=False)
    day_number = Column(Integer, nullable=False) # Day 1, Day 2, etc.
    name = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    
    # Category: hotel, breakfast, lunch, dinner, museum, event, attraction, coffee, transit, rest, etc.
    type = Column(String(100), nullable=False) 
    
    # Store times as standard 'HH:MM' string for SQLite / Postgres portability
    start_time = Column(String(10), nullable=True) 
    end_time = Column(String(10), nullable=True) 
    
    # Location coordinates & addresses
    latitude = Column(Float, nullable=True)
    longitude = Column(Float, nullable=True)
    address = Column(String(500), nullable=True)
    
    # Contact assistance info
    contact_phone = Column(String(100), nullable=True)
    contact_website = Column(String(255), nullable=True)
    contact_email = Column(String(255), nullable=True)
    
    # Budget optimization info
    cost = Column(Float, default=0.0)
    rating = Column(Float, nullable=True)
    
    # Intelligent transit planning details from previous activity (or airport/station)
    travel_distance_km = Column(Float, nullable=True)
    travel_duration_min = Column(Float, nullable=True)
    travel_mode = Column(String(50), nullable=True) # driving, transit, walking, bicycling
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    # Relationships
    trip = relationship("Trip", back_populates="activities")

File: backend/src/main/java/com/conciergeiq/model/Activity.java
package com.conciergeiq.model;

import jakarta.persistence.*;
import lombok.*;
import java.util.UUID;

@Entity
@Table(name = "activities")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Activity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "trip_id", nullable = false)
    private Trip trip;

    @Column(name = "day_number", nullable = false)
    private Integer dayNumber;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String type; // hotel, breakfast, lunch, dinner, museum, event, attraction, transit, rest, coffee

    @Column(name = "start_time")
    private String startTime; // HH:MM

    @Column(name = "end_time")
    private String endTime; // HH:MM

    private Double latitude;
    private Double longitude;
    private String address;

    @Column(name = "contact_phone")
    private String contactPhone;

    @Column(name = "contact_website")
    private String contactWebsite;

    private Double cost = 0.0;

    @Column(name = "travel_distance_km")
    private Double travelDistanceKm;

    @Column(name = "travel_duration_min")
    private Double travelDurationMin;

    @Column(name = "travel_mode")
    private String travelMode;

    @Column(name = "booking_confirmation_code")
    private String bookingConfirmationCode;
}
